crates/bridge-exec/src/ws_client.rs

```diff
@@ -0,0 +1,159 @@
+            .build(self.config.url.clone())
+            .await;
+        let bl = match client {
+            Ok(cl) => WsClientState::Working(cl),
+            Err(_) => WsClientState::NotWorking,
+        };
+        Ok(WsClient(bl))
+    }
+
+    /// Recycles an existing WebSocket client.
+    ///
+    /// Checks whether the provided client is still valid. If the client is connected,
+    /// it is retained. Otherwise, an error is returned, indicating the need to recreate
+    /// the client.
+    async fn recycle(
+        &self,
+        obj: &mut Self::Type,
+        _metrics: &managed::Metrics,
+    ) -> RecycleResult<Self::Error> {
+        match &obj.0 {
+            WsClientState::Working(cl) => {
+                if cl.is_connected() {
+                    Ok(())
+                } else {
+                    Err(RecycleError::Message(
+                        "Connection lost, recreate client".to_string().into(),
+                    ))
+                }
+            }
+            WsClientState::NotWorking => Err(RecycleError::Message(
+                "Connection still not found, recreate client"
+                    .to_string()
+                    .into(),
+            )),
+        }
+    }
+}
+
+/// Implements the `ClientT` trait for `WsClient`.
+///
+/// This implementation allows `WsClient` to perform JSON-RPC operations,
+/// including notifications, method calls, and batch requests.
+#[async_trait]
+impl ClientT for WsClient {
+    /// Sends a notification request.
+    async fn notification<Params>(&self, method: &str, params: Params) -> Result<(), ClientError>
+    where
+        Params: ToRpcParams + Send,
+    {
+        match &self.0 {
+            WsClientState::Working(inner) => inner.notification(method, params).await,
+            WsClientState::NotWorking => Err(ClientError::Transport(BoxError::from(
+                "Client is Not Working".to_string(),
+            ))),
+        }
+    }
+
+    /// Sends a method call request.
+    async fn request<R, Params>(&self, method: &str, params: Params) -> Result<R, ClientError>
+    where
+        R: DeserializeOwned,
+        Params: ToRpcParams + Send,
+    {
+        match &self.0 {
+            WsClientState::Working(inner) => inner.request(method, params).await,
+            WsClientState::NotWorking => Err(ClientError::Transport(BoxError::from(
+                "Client is Not Working".to_string(),
+            ))),
+        }
+    }
+
+    /// Sends a batch request.
+    async fn batch_request<'a, R>(
+        &self,
+        batch: BatchRequestBuilder<'a>,
+    ) -> Result<BatchResponse<'a, R>, ClientError>
+    where
+        R: DeserializeOwned + fmt::Debug + 'a,
+    {
+        match &self.0 {
+            WsClientState::Working(inner) => inner.batch_request(batch).await,
+            WsClientState::NotWorking => Err(ClientError::Transport(BoxError::from(
+                "Client is Not Working".to_string(),
+            ))),
```

Comment:
As a cleanliness thing you could have a function like `make_notworking_err` to dedup constructing this complex error instance.